src/systemctl/systemctl.c

```diff
@@ -3231,16 +3231,16 @@ static int logind_check_inhibitors(enum action a) {
         char **s;
         int r;
 
-        if (arg_ignore_inhibitors || arg_force > 0)
+        if (streq(arg_respect_inhibitors, "no") || arg_force > 0)
```

Comment:
please parse that in getopt() handler already, as mentioned above.

Also, please parse this with parse_boolean(), which is how we generally handle stuff that is a boolean or boolean like. i.e.: 
```c
…
case ARG_RESPECT_INHIBITORS:
        if (streq(optarg, "auto"))
                arg_respect_inhibitors = -1;
        else {
                r = parse_boolean(optarg);
                if (r < 0)
                        return log_error_errno(r, "Failed to parse --respect-inhibitors= argument: %s", optarg);
                arg_respect_inhibitors = r;
        }
        break;
…
```